### PR TITLE
Smorgasboard of doc improvements

### PR DESCRIPTION
**Pull request by alexcrichton**
* Add deep dive documentation for `tests/web.rs` of the `wasm-pack-template`
* Add docs for testing a project in addition to building
* Touch up other documentation int he deep dive and setup
* Fix a few dead links and moved pages and such

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
rebuilding the appveyor build because it failed for a seemingly spurious reason.

File: docs/src/tutorials/npm-browser-packages/template-deep-dive/testing-your-project.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Testing your project
```
Comment (ashleygwilliams) [inline docs/src/tutorials/npm-browser-packages/template-deep-dive/testing-your-project.md]: this looks like a stub that mdbook created - might mean your file is in the wrong directory?

Comment (alexcrichton) [reply docs/src/tutorials/npm-browser-packages/template-deep-dive/testing-your-project.md]: Oops yes, this shouldn't be here, deleted now

**Comment by ashleygwilliams** [comment]
🎉

**Pull request merged**